pinot-udf/src/main/java/org/apache/pinot/scalar/function/hypertrace/HypertraceScalarFunctions.java

```diff
@@ -24,6 +28,17 @@ public static String defaultString(String value, String defaultValue) {
             replaceNullStringWithNull(value), replaceNullStringWithNull(defaultValue)));
   }
 
+  @ScalarFunction(name = STRING_EQUALS_FUNCTION_NAME)
+  public static String stringEquals(String s1, String s2) {
+    return String.valueOf(Equals.stringEquals(s1, s2));
+  }
+
+  @ScalarFunction(name = CONDITIONAL_FUNCTION_NAME)
+  public static String conditional(String condition, String s1, String s2) {
+    Boolean conditionBooleanVal = Boolean.getBoolean(condition);
```

Comment:
The docs of this method make me suspect it's not what we're looking for:
```
     * Returns {@code true} if and only if the system property named
     * by the argument exists and is equal to, ignoring case, the
     * string {@code "true"}.
```

parseBoolean is probably closer, but we also want to check for `NULL_STRING` and return `NULL_STRING` if needed (in both methods)

Reply:
Switched to parseBoolean. You also want to check if the return value for stringEquals is `null` or should we assume that it will always `true` or `false`?